state/modelgeneration.go

```diff
@@ -635,6 +635,23 @@ func (m *Model) Branch(name string) (*Generation, error) {
 	return gen, errors.Trace(err)
 }
 
+func (m *Model) applicationBranches(appName string) ([]*Generation, error) {
+	branches, err := m.Branches()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	foundBranches := make([]*Generation, 0)
+	for _, branch := range branches {
```

Comment:
Suggestion:
Add `IsTrackedBy(unit)` and `HasChangesFor(application)` for to `Generation` for modularisation and decreased loop/if nesting.

Reply:
Added generation.IsTracking() to pr 10415.  Added generation.HasChangesFor() to this one.